testsuite/smoke-test/src/aptos_cli/governance.rs

```diff
@@ -0,0 +1,241 @@
+            // enough for quorum
+            *genesis_stake_amount = 5000000;
+        }))
+        .with_init_genesis_config(Arc::new(|genesis_config| {
+            genesis_config.allow_new_validators = true;
+            genesis_config.epoch_duration_secs = 5;
+            // lockup duration need to be large enough to vote
+            genesis_config.recurring_lockup_duration_secs = 100;
+            // for early proposal execution
+            genesis_config.voting_duration_secs = 1;
+            genesis_config.min_stake = MIN_STAKE;
+        }))
+        .build_with_cli(0)
+        .await;
+
+    let transaction_factory = swarm.chain_info().transaction_factory();
+    let rest_client = swarm.validators().next().unwrap().rest_client();
+
+    let mut keygen = KeyGen::from_os_rng();
+
+    let owner_initial_coins = 1100000;
+    let voter_initial_coins = 10000;
+    let operator_initial_coins = 10000;
+
+    // Owner of the coins receives coins
+    let owner_cli_index = cli
+        .create_cli_account_from_faucet(
+            keygen.generate_ed25519_private_key(),
+            Some(owner_initial_coins),
+        )
+        .await
+        .unwrap();
+
+    // faucet can make our root LocalAccount sequence number get out of sync.
+    swarm
+        .chain_info()
+        .resync_root_account_seq_num(&rest_client)
+        .await
+        .unwrap();
+
+    let operator_keys = ValidatorNodeKeys::new(&mut keygen);
+    let voter_cli_index = cli
+        .create_cli_account(keygen.generate_ed25519_private_key(), owner_cli_index)
+        .await
+        .unwrap();
+    let operator_cli_index = cli
+        .create_cli_account(operator_keys.account_private_key.clone(), owner_cli_index)
+        .await
+        .unwrap();
+
+    cli.transfer_coins(owner_cli_index, voter_cli_index, voter_initial_coins, None)
+        .await
+        .unwrap();
+    cli.transfer_coins(
+        owner_cli_index,
+        operator_cli_index,
+        operator_initial_coins,
+        None,
+    )
+    .await
+    .unwrap();
+
+    let stake_amount = 1000000;
+    cli.initialize_stake_owner(
+        owner_cli_index,
+        stake_amount,
+        Some(voter_cli_index),
+        Some(operator_cli_index),
+    )
+    .await
+    .unwrap();
+
+    let port = 6543;
+
+    cli.update_consensus_key(
+        operator_cli_index,
+        Some(owner_cli_index),
+        operator_keys.consensus_public_key(),
+        operator_keys.consensus_proof_of_possession(),
+    )
+    .await
+    .unwrap();
+
+    cli.update_validator_network_addresses(
+        operator_cli_index,
+        Some(owner_cli_index),
+        HostAndPort {
+            host: DnsName::try_from("0.0.0.0".to_string()).unwrap(),
+            port,
+        },
+        operator_keys.network_public_key(),
+    )
+    .await
+    .unwrap();
+
+    cli.join_validator_set(operator_cli_index, Some(owner_cli_index))
+        .await
+        .unwrap();
+
+    reconfig(
```

Comment:
nit: Add comment to explain we wait for next epoch so the validator is in the active validator set.

Reply:
This test could halt if the other validator doesn't have enough stake.  Might need to add 4 validators to run. @zekun000